src/Bundle/Resources/config/services/state.xml

```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+
+ This file is part of the Sylius package.
+
+ (c) Paweł Jędrzejewski
+
+ For the full copyright and license information, please view the LICENSE
+ file that was distributed with this source code.
+
+-->
+
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <service id="sylius.state.provider" class="Sylius\Component\Resource\State\Provider">
+            <argument type="tagged_locator" tag="sylius.state_provider" />
```

Comment:
I'm afraid, these tagged locators are broken because we needed to define a key that will be taken into account. What is more, I would recommend adding a default indexing method and adding them as an optional method (as a comment above the interface). Did you test them out?